crates/evm/core/src/backend/mod.rs

```diff
@@ -1892,25 +1892,29 @@ fn commit_transaction<I: Inspector<Backend>>(
     Ok(())
 }
 
+/// Helper method which updates data in the state with the data from the database.
+pub fn update_state<DB: Database>(state: &mut State, db: &mut DB)
+where
+    DB::Error: core::fmt::Debug,
+{
+    for (addr, acc) in state.iter_mut() {
+        acc.info = db.basic(*addr).unwrap().unwrap_or_default();
```

Comment:
no unwraps on db operations, 
unclear if this is even reachable, but we should handle the error gracefully